app/models/stage.rb

```diff
@@ -98,7 +98,7 @@ def confirm_before_deploying?
   end
 
   def create_deploy(user, attributes = {})
-    deploys.create(attributes) do |deploy|
+    deploys.create(attributes.merge(release: !no_code_deployed)) do |deploy|
```

Comment:
Why would we need to pass in the `:no_code_deployed` ... a stage is always the same command, so it's always either a deploy or not a deploy